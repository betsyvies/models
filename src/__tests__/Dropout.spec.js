const mongoose = require('mongoose');
const { User, Dropout } = require('../../')(mongoose);

describe('Dropout', () => {
  let user;

  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__, { useNewUrlParser: true });
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  beforeEach(async () => {
    await mongoose.connection.db.dropDatabase();
    await Dropout.createIndexes();
    await User.createIndexes();
    user = new User({
      uid: 'abcdefghijklmnopqrstuvwxyz',
      email: 'someone@somewhere.com',
      name: 'Someone',
    });
    await user.save();
  });

  it('should fail when user is not an ObjectId', () => {
    const doc = new Dropout({
      city: 'LIM',
      cohort: 'LIM007',
      user: 'user._id',
      stage: 'Project 1',
      studentCode: 'LIM181080',
      when: '5/14/2018',
      type: 'Leave the program voluntarily',
      reason: 'Problemas familiares y económicos',
      observations: 'Una lástima, era una buena estudiante :(',
      sad: true,
      otherReason: 'I do not know (pero ya agoté todos los recursos para averiguarlo)',
    });

    return doc.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should fail when user does not exist', async () => {
    const doc = new Dropout({
      city: 'LIM',
      cohort: 'LIM007',
      user: (new User())._id,
      stage: 'Project 1',
      studentCode: 'LIM181080',
      when: '5/14/2018',
      type: 'Leave the program voluntarily',
      reason: 'Problemas familiares y económicos',
      observations: 'Una lástima, era una buena estudiante :(',
      sad: true,
      otherReason: 'I do not know (pero ya agoté todos los recursos para averiguarlo)',
    });

    return doc.save()
      .catch((err) => {
        expect(err.message).toBe('user does not exist');
      });
  });
  it('should create a new dropout', async () => {
    const doc = new Dropout({
      city: 'LIM',
      cohort: 'LIM007',
      user: user._id,
      stage: 'Project 1',
      studentCode: 'LIM181080',
      when: '5/14/2018',
      type: 'Leave the program voluntarily',
      reason: 'Problemas familiares y económicos',
      observations: 'Una lástima, era una buena estudiante :(',
      sad: true,
      otherReason: 'I do not know (pero ya agoté todos los recursos para averiguarlo)',
    });

    return doc.save()
      .then((dropout) => {
        expect(dropout.user).toBe(user._id);
      });
  });
});
