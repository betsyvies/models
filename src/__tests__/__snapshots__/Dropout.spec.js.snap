// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropout should fail when user is not an ObjectId 1`] = `
Object {
  "user": [CastError: Cast to ObjectID failed for value "user._id" at path "user"],
}
`;
